### PR TITLE
add script to make hdf data cache, allow it to be read by cli tools. 

### PR DESCRIPTION
**Pull request by ahnitz**
@titodalcanton Also, pinging you as in the process of using this I encountered a bug in the 'detect_loud_stuff" function for doing gating. It seems if it actually does a resampling it wasn't handling the corruption padding correctly later on.

### CODE REVIEW

**Comment by ahnitz** [comment]
Example command line for making an hdf data store.

```
pycbc_condition_hdf --science-name '+DATA' --instrument H1 --frame-type LOSC --channel-name H1:GWOSC-16KHZ_R1_STRAIN --gps-start-time 1187007882 --gps-end-time 1187009882 --sample-rate 2048 --pad-data 8 --output-file H1.hdf --verbose --strain-high-pass 15.0
```

**Comment by spxiwh** [comment]
This isn't passing Travis at the moment, and CodeClimate is complaining about poorly named variables. I'll take a proper look over once these things are fixed.

**Comment by ahnitz** [comment]
@spxiwh I think this should be good now.

File: bin/pycbc_condition_hdf [hunk 1]
```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python
```
Comment (spxiwh) [inline bin/pycbc_condition_hdf]: It seems weird to have a `pycbc_condition_strain` and a `pycbc_condition_hdf`, both doing very similar things. Can these not be combined?

File: pycbc/frame/__init__.py [hunk 2]
```diff
@@ -2,6 +2,8 @@
                      query_and_read_frame, frame_paths, write_frame,
                      DataBuffer, StatusBuffer)
 
+from . store import (read_store)
```
Comment (spxiwh) [inline pycbc/frame/__init__.py]: Isn't that space not supposed to be there?

Comment (spxiwh) [reply pycbc/frame/__init__.py]: `from .store import read_store`?

File: bin/pycbc_condition_hdf (hunk 1 above)
Comment (spxiwh) [reply bin/pycbc_condition_hdf]: (Maybe these are sufficiently different as `condition_hdf` is writing a large data set ... Maybe the executable name is the problem ...)

File: pycbc/frame/__init__.py (hunk 2 above)
Comment (ahnitz) [reply pycbc/frame/__init__.py]: Either works. I'm not sure which is the preferred style.... Maybe .store is more common?

File: pycbc/strain/strain.py [hunk 3]
```diff
@@ -93,20 +93,17 @@ def detect_loud_glitches(strain, psd_duration=4., psd_stride=2.,
...
+
     logging.info('Autogating: tapering strain')
     taper_length = int(corrupt_time * strain.sample_rate)
     w = numpy.arange(taper_length) / float(taper_length)
     strain[0:taper_length] *= pycbc.types.Array(w, dtype=strain.dtype)
     strain[(len(strain)-taper_length):] *= pycbc.types.Array(w[::-1],
                                                              dtype=strain.dtype)
-
-    # don't waste time trying to optimize a single FFT
-    pycbc.fft.fftw.set_measure_level(0)
```
Comment (spxiwh) [inline pycbc/strain/strain.py]: Why remove this?

File: bin/pycbc_condition_hdf (hunk 1 above)
Comment (ahnitz) [reply bin/pycbc_condition_hdf]: Ok, that makes sense. Maybe pycbc_data_store?

Comment (ahnitz) [reply bin/pycbc_condition_hdf]: Ok, I've changed the file name

**Pull request merged**